libs/queue/README.md

```diff
@@ -0,0 +1,64 @@
+# Overview
+
+The Queue library provides an implementation of a queue data structure. A Queue is a linear structure which follows the First-In-First-Out (FIFO) principle. This means that the elements added first are the ones that get removed first.
+
+The `Queue<T>` is a structure with a generic type parameter T. The elements in the queue are stored as a `Vec<T>`. The data structure is created empty and is growable.
+
+For more information please see the [specification](./SPECIFICATION.md).
+
+> **Note** The Queue<T> type in this library is different from built-in arrays and vectors due to its unique characteristics, especially its operation in FIFO manner.
+
+## Known Issues
+
+Currently, the Queue<T> type does not provide any way to iterate through its elements. 
```

Comment:
I'd be weary of listing items that may be feature requests as opposed to issues.

For example, if the queue usually provides the feature for iteration but under some concrete type for `T` the iteration breaks then that is a known issue.
On the other hand, if the feature is entirely missing then that's not an issue, it's a missing feature.

Moreover, I would consider whether there are additional complications with the implementation.

Reply:
Got it, will remove it from Known Issues.